node_modules/col-assets/lib/api.js

```diff
@@ -1008,6 +1008,16 @@ var canDeleteAsset = function(user, asset) {
   }
 };
 
+/**
+ * Returns true if the asset is not hidden, not deleted, and not associated with only hidden categories.
+ *
+ * @param   {Asset}        asset             The asset to check for deleted or hidden status
+ * @return  {boolean}                        True if deleted or hidden
+ */
+var isDeletedOrHidden = module.exports.isDeletedOrHidden = function(asset) {
+  return !asset.visible || !!asset.deleted_at || !(_.isEmpty(asset.categories) || _.find(asset.categories, 'visible'));
```

Comment:
@johncrossman Oh, you're quite right - that comment is secret evidence that the method was originally written the other way round.